packages/playback-core/src/index.ts

```diff
@@ -1251,8 +1251,68 @@ export const loadMedia = (
     });
 
     hls.on(Hls.Events.ERROR, (_event, data) => {
-      saveAndDispatchError(mediaEl, getErrorFromHlsErrorData(data, props));
+      const error = getErrorFromHlsErrorData(data, props);
+
+      if (error.muxCode === MuxErrorCode.NETWORK_NOT_READY) {
+        const maxRetries = 6; // 5 minutes and 5 seconds total (5s, 60s, 60s, 60s, 60s, 60s)
+        const state = muxMediaState.get(mediaEl) ?? {};
+        const retryCount = state.retryCount ?? 0;
+
+        if (retryCount < maxRetries) {
+          // First retry is 5 seconds, subsequent retries are 60 seconds
+          const retryDelay = retryCount === 0 ? 5000 : 60000;
+
+          // New error with the retry delay
+          const retryDelayError = new MediaError(
+            `Retrying in ${retryDelay / 1000} seconds...`,
+            error.code,
+            error.fatal
+          );
+          Object.assign(retryDelayError, error);
+          saveAndDispatchError(mediaEl, retryDelayError);
+
+          setTimeout(() => {
+            state.retryCount = retryCount + 1;
+            if (data.details === 'manifestLoadError' && data.url) {
+              hls.loadSource(data.url);
+            }
+          }, retryDelay);
+          return;
+        } else {
+          state.retryCount = 0;
+          // New error with the retry link
+          const retryLinkError = new MediaError(
+            'Try again later or <a href="#" onclick="window.location.reload(); return false;" style="color: #4a90e2;">click here to retry</a>',
+            error.code,
+            error.fatal
+          );
+          Object.assign(retryLinkError, error);
+          saveAndDispatchError(mediaEl, retryLinkError);
+          return;
+        }
+      }
+      saveAndDispatchError(mediaEl, error);
     });
+
+    hls.on(Hls.Events.MANIFEST_LOADED, () => {
+      // Clear error state and UI
+      const state = muxMediaState.get(mediaEl);
+      if (state) {
+        state.error = null;
+        state.retryCount = 0;
+      }
+
+      const clearError = new MediaError('', MediaError.MEDIA_ERR_CUSTOM, false);
+      clearError.muxCode = MuxErrorCode.NOT_AN_ERROR;
+      mediaEl.dispatchEvent(
+        new CustomEvent('error', {
+          detail: clearError,
+          composed: true,
+          bubbles: true,
+        })
+      );
```

Comment:
we should find something else to clear the error dialog. 
this is too hacky :)

I don't love this either but try dispatching an emptied followed by an loadstart event. that feels more inline with what's happening and media-chrome listens to emptied to change media error state.

Try without composed and bubbles, that is only needed for non native video events. if custom-media-element is used, it forwards native video events from the native video element. https://github.com/muxinc/media-elements/blob/45b7383d445d2969409ec369aa34e967c4d32cab/packages/custom-media-element/custom-media-element.ts#L345-L359

Like this:

```js
mediaEl.dispatchEvent(new Event('emptied'));
mediaEl.dispatchEvent(new Event('loadstart'));
```